discojs/discojs-core/src/models/gpt/index.ts

```diff
@@ -91,6 +57,10 @@ export class GPT extends Model {
     for (let i = 0; i < epochs; i++) {
       await this.model.fitDataset(trainingData, trainingArgs)
       yield logs
+      if (logs !== undefined && logs.training_loss < 0.01) {
+        console.log("Early stopping")
+        break
+      }
```

Comment:
let's users control when to stop, maybe they want even less training_loss, or stop it for wathever reasons.
when using a generator, one [can stop it using `generator.return()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator/return), or simply never read the next value (will resolve it's promise still but not advance anymore).

```suggestion
```